### PR TITLE
Refactor CollapseSection

### PR DESCRIPTION
**Pull request by kulyk**
This PR refactors the `CollapseSection` component, so it aligns with our current FE code style. Prop types, styled components, unit tests, etc.

On top of that, it adds two new props:

* `iconVariant: "right-down" | "up-down"` — defines which chevron icons to use for expanded and collapsed states
* `iconPosition: "left" | "right"` — defines where the icon should be rendered relative to the section header

FYI, the component had a lot of class names, but I couldn't find them anywhere in the project, seems to be a legacy.

### CODE REVIEW

File: frontend/src/metabase/components/CollapseSection/CollapseSection.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable react/display-name */
```
Comment (gusaiani) [inline frontend/src/metabase/components/CollapseSection/CollapseSection.unit.spec.js:1]: 🤔 Do we need this eslint disabler?

File: frontend/src/metabase/components/CollapseSection/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default } from "./CollapseSection";
```
Comment (gusaiani) [inline frontend/src/metabase/components/CollapseSection/index.js:1]: 🤔 I haven't been doing this, I've been importing from URIs with dir and filename like `path/to/CollapseSection/CollapseSection`

We've touched on this in a RFC.

I don't know, I just think it's one fewer file, and generally less indirection.
But happy to follow any of the patterns.

**Review by gusaiani — APPROVED** [review]
Looking great, just very minor comments.

File: frontend/src/metabase/components/CollapseSection/CollapseSection.unit.spec.js (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/components/CollapseSection/CollapseSection.unit.spec.js:1]: Yeah, there is a functional component below in the header prop test and eslint would highlight it. Suppose we should maybe disable some rules for the test files

**Pull request merged**